Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPISidebar.mm

```diff
@@ -219,6 +225,514 @@
     Util::loadAndRunExtension(sidebarActionManifest, @{ @"background.js": Util::constructScript(script) }, sidebarConfig);
 }
 
+TEST_F(WKWebExtensionAPISidebar, SidebarActionAPIGlobalTitlePersists)
+{
+    auto *script = @[
+        @"await browser.sidebarAction.setTitle({ title: 'Here is a title' })",
+        @"browser.test.assertDeepEq(await browser.sidebarAction.getTitle({}), 'Here is a title')",
+        @"browser.test.notifyPass()",
+    ];
+
+    Util::loadAndRunExtension(sidebarActionManifest, @{ @"background.js": Util::constructScript(script) }, sidebarConfig);
+}
+
+TEST_F(WKWebExtensionAPISidebar, SidebarActionAPIWindowSpecificTitlePersists)
+{
+    auto *script = @[
+        @"let windows = await browser.windows.getAll()",
+        @"let [window1, window2] = windows",
+
+        @"await browser.sidebarAction.setTitle({ windowId: window1.id, title: 'window-specific title' })",
+        @"await browser.sidebarAction.setTitle({ title: 'global title' })",
+
+        @"browser.test.assertDeepEq(await browser.sidebarAction.getTitle({ windowId: window1.id }), 'window-specific title')",
+        @"browser.test.assertDeepEq(await browser.sidebarAction.getTitle({}), 'global title')",
+        @"browser.test.assertDeepEq(await browser.sidebarAction.getTitle({ windowId: window2.id }), 'global title')",
```

Comment:
`browser.test.assertEq` is better for strings and non-object types, unless you had another reason for `assertDeepEq`? `assertDeepEq` is kind of a hack currently…
```suggestion
        @"browser.test.assertEq(await browser.sidebarAction.getTitle({ windowId: window1.id }), 'window-specific title')",
        @"browser.test.assertEq(await browser.sidebarAction.getTitle({}), 'global title')",
        @"browser.test.assertEq(await browser.sidebarAction.getTitle({ windowId: window2.id }), 'global title')",
```

Reply:
I'll go ahead and switch all of those over -- I only used `assertDeepEq` to stay consistent with my habit of always using `===` instead of `==` when writing JS.